plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Marketing.php

```diff
@@ -125,13 +125,13 @@ public static function has_installed_extensions() {
 			}
 		}
 
-		// All extensions installed.
-		if ( count( $remaining ) === 0 ) {
-			return true;
-		}
-
-		// Make sure the task has been actioned and at least one extension is installed.
-		if ( count( $installed ) > 0 && Task::is_task_actioned( 'marketing' ) ) {
+		// Make sure the task has been actioned or all extensions installed
+		// or a marketing extension has been installed.
+		if (
+			count( $installed ) > 0 ||
+			Task::is_task_actioned( 'marketing' ) ||
+			count( $remaining ) === 0
```

Comment:
Just curious, this condition seems a little redundant with the first one, since `$remaining` is just the inverse of `$installed`, but perhaps it's needed to capture the condition that there are _no_ plugins?

Reply:
Nice catch! I removed the redundant condition in commit 48a3724